### ISSUE TITLE
KafkaActorSubscriber log statement not displaying exception

### ISSUE THREAD

**Issue by gvd** [softwaremill/reactive-kafka#21]
In https://github.com/softwaremill/reactive-kafka/blob/master/src/main/scala/com/softwaremill/react/kafka/KafkaActorSubscriber.scala on line 36 I see:

log.error("Stopping subscriber due to an error", ex)

Which does not display the exception message

**Comment by kciesielski** [comment]
Thanks, @gvd! Indeed, the logging interface is a bit unfortunate, I'm fixing this to 
`log.error(ex, "Stopping subscriber due to an error")`
I'll try to add some test around it.

**Comment by adamw** [comment]
Maybe you could have an `ExceptionHandler` component, where the default handling would be to log the exceptions, but could provide hooks to e.g. restart the whole stream, try to reconnect, do exponential backoff etc?
Passed as an implicit or parameter with a default value?

**Comment by kciesielski** [comment]
@adamw Underneath the subscriber is an Akka actor. Errors can be handled with custom behavior using DeathWatch. I'm checking how one can pass more details on exceptions with that.

**Comment by xelax** [comment]
I might be wrong, but currently the underlying actor is not returned when build the stream, thus inhibiting the ability to supervise/register for deathwatch of that actor.

**Comment by kciesielski** [comment]
There is some public API not mentioned in README. You can build your consumer like this:
```scala
val consumerActor = reactiveKafka.consumerAsActor[T](...)
val consumer = ActorPublisher[String](consumerActor)
```
and you have a working ActorRef which you can watch. I'm also trying to make it possible to supervise this actor, so that you can capture full exception when it crashes. This way

**Comment by kciesielski** [comment]
Fixed flly in #23
